src/client/Microsoft.Identity.Client/ApiConfig/BrokerOptions.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Microsoft.Identity.Client.ApiConfig
+{
+    /// <summary>
+    /// The class specifies the options for broker across OperatingSystems
+    /// The common properties are direct members
+    /// Platform specific properties (if they exist) are part of the corresponding options
+    /// </summary>
+    public class BrokerOptions
+    {
+        /// <summary>
+        /// Supported OperatingSystems
+        /// </summary>
+        [Flags]
+        public enum OperatingSystems
+        {
+            /// <summary>
+            /// No OS specified - Invalid options
+            /// </summary>
+            None = 0b_0000_0000,  // 0
+            /// <summary>
+            /// Use broker on Windows OS
+            /// </summary>
+            Windows = 0b_0000_0001,  // 1
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="osChoices">Choices of OperatingSystems</param>
+        /// <param name="title">Title of the broker</param>
+        /// <param name="msaPassthrough">is this MsaPasstorugh</param>
+        public BrokerOptions(OperatingSystems osChoices, string title = "", bool msaPassthrough = false)
+        {
+            if (osChoices == OperatingSystems.None)
+            {
+                throw new ArgumentException($"osChoices should not be none");
+            }
+
+            OSChoices = osChoices;
+            Title = title;
+            MsaPassthrough = msaPassthrough;
+        }
+
+        // The default constructor is private. So developer is forced to set the OS choice(s)
+        private BrokerOptions()
+        {
+
+        }
+
+        /// <summary>
+        /// Creates default options that can be modified later except the choice of OS
+        /// </summary>
+        /// <param name="osChoice">Choice of OS platforms</param>
+        /// <param name="listWorkAndSchoolAccts">List wokr and school accounts</param>
+        /// <returns></returns>
+        public static BrokerOptions CreateDefault(OperatingSystems osChoice = OperatingSystems.Windows, bool listWorkAndSchoolAccts = true)
+        {
+            BrokerOptions ret = new BrokerOptions(osChoice);
+            var winBrokerDefaultOptions = WindowsBrokerOptions.CreateDefault();
+            ret.Title = winBrokerDefaultOptions.HeaderText;
+            ret.MsaPassthrough = winBrokerDefaultOptions.MsaPassthrough;
+            ret.ListOperatingSystemAccounts = listWorkAndSchoolAccts;
+            
+            return ret;
+        }
+
+        /// <summary>
+        /// Creates BrokerOptions from WindowsBrokerOptions
+        /// </summary>
+        /// <param name="winOptions"></param>
+        /// <param name="osChoice"></param>
+        /// <returns></returns>
+        public static BrokerOptions CreateFromWindowsOptions(WindowsBrokerOptions winOptions, OperatingSystems osChoice = OperatingSystems.Windows)
```

Comment:
Should be internal? Users should just create their own BrokerOptions instance, the properties are basically the same.

```suggestion
        internal static BrokerOptions CreateFromWindowsOptions(WindowsBrokerOptions winOptions, OperatingSystems osChoice = OperatingSystems.Windows)
```